insights/attachments/links_qr_contains_recipient_email.yml

```
@@ -2,18 +2,19 @@ name: "QR code link in attachment contains recipient email address"
 type: "query"
 source: |
   map(filter(attachments,
-             .file_type in $file_types_images
-             or .file_extension in $file_extensions_macros
-             or .file_type == "pdf"
-      ),
-      map(filter(file.explode(.),
-                 .scan.qr.type == "url"
-                 and any(recipients.to,
-                         strings.icontains(..scan.qr.url.url, .email.email)
-                         and .email.domain.valid
-                 )
-          ),
-          .scan.qr.url.url
-      )
-  )
+               .file_type in $file_types_images
+               or .file_extension in $file_extensions_macros
+               or .file_type == "pdf"
+        ),
+        map(filter(file.explode(.),
+                   .scan.qr.type == "url"
+                   and any(recipients.to,
+                           (strings.icontains(..scan.qr.url.url, .email.email)
+                           and .email.domain.valid) or any(beta.scan_base64(..scan.qr.url.url),
+                      strings.icontains(., ..email.email))
+                   )
```

apply .email.domain.valid first
```suggestion
                   and any(recipients.to,
                           .email.domain.valid
                           and (
                             strings.icontains(..scan.qr.url.url, .email.email)
                             or any(beta.scan_base64(..scan.qr.url.url),
                               strings.icontains(., ..email.email)
                             )
                   )
```

---

could probably benefit from a formatting operation as well.